libkrb5/src/credential.rs

```diff
@@ -151,28 +151,44 @@ impl<'a> Drop for Krb5Creds<'a> {
     }
 }
 
-#[derive(Clone)]
-pub struct Krb5Keyblock {
-    magic: krb5_magic,
-    pub(crate) enctype: krb5_enctype,
-    pub(crate) contents: Vec<u8>,
+pub struct Krb5Keyblock<'a> {
+    pub(crate) context: &'a Krb5Context,
+    pub(crate) keyblock: &'a mut krb5_keyblock,
 }
 
-impl Krb5Keyblock {
-    pub fn from_c(raw_keyblock: &krb5_keyblock) -> Krb5Keyblock {
-        Krb5Keyblock {
-            magic: raw_keyblock.magic,
-            enctype: raw_keyblock.enctype,
-            contents: unsafe { slice::from_raw_parts(raw_keyblock.contents, raw_keyblock.length as usize).to_vec() },
+impl<'a> Drop for Krb5Keyblock<'a> {
+    fn drop(&mut self) {
+        unsafe {
+            krb5_free_keyblock(self.context.context, self.keyblock);
         }
     }
-    pub fn to_c(&mut self) -> *mut krb5_keyblock {
-        let keyblock = krb5_keyblock {
-            magic: self.magic,
-            enctype: self.enctype,
-            length: self.contents.len() as u32,
-            contents: self.contents.as_mut_ptr(),
+}
+
+impl<'a> Krb5Keyblock<'a> {
+    pub fn copy(&self) -> Result<Self, Krb5Error> {
+        let mut keyblock_ptr: MaybeUninit<*mut krb5_keyblock> = MaybeUninit::zeroed();
```

Comment:
Nem tudná a copy behívni a new_from_rawt(self.context, self.keyblock)-kal, vagy akár össze is vonni a kettőt, mert igazából mindekettő egy dup() lesz.

Reply:
Done